codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/PythonType.kt

```diff
@@ -0,0 +1,174 @@
+    }
+
+    data class Set(override val member: PythonType) : PythonType(), Container {
+        override val name: String = "Set"
+        override val namespace: String = "typing"
+    }
+
+    data class Optional(override val member: PythonType) : PythonType(), Container {
+        override val name: String = "Optional"
+        override val namespace: String = "typing"
+    }
+
+    data class Awaitable(override val member: PythonType) : PythonType(), Container {
+        override val name: String = "Awaitable"
+        override val namespace: String = "typing"
+    }
+
+    data class Callable(val args: kotlin.collections.List<PythonType>, val rtype: PythonType) : PythonType() {
+        override val name: String = "Callable"
+        override val namespace: String = "typing"
+    }
+
+    data class Union(val args: kotlin.collections.List<PythonType>) : PythonType() {
+        override val name: String = "Union"
+        override val namespace: String = "typing"
+    }
+
+    data class Opaque(override val name: String, val rustNamespace: String? = null) : PythonType() {
+        // Since Python doesn't have a something like Rust's `crate::` we are using a custom placeholder here
+        // and in our stub generation script we will replace placeholder with the real root module name.
+        private val pythonRootModulePlaceholder = "__root_module_name__"
+
+        override val namespace: String? = rustNamespace?.split("::")?.joinToString(".") {
+            when (it) {
+                "crate" -> pythonRootModulePlaceholder
+                // In Python, we expose submodules from `aws_smithy_http_server_python`
+                // like `types`, `middleware`, `tls` etc. from `__root_module__name`
+                "aws_smithy_http_server_python" -> pythonRootModulePlaceholder
+                else -> it
+            }
+        }
+    }
+}
+
+/**
+ * Return corresponding [PythonType] for a [RustType].
+ */
+fun RustType.pythonType(): PythonType =
+    when (this) {
+        is RustType.Unit -> PythonType.None
+        is RustType.Bool -> PythonType.Bool
+        is RustType.Float -> PythonType.Float
+        is RustType.Integer -> PythonType.Int
+        is RustType.String -> PythonType.Str
+        is RustType.Vec -> PythonType.List(this.member.pythonType())
+        is RustType.Slice -> PythonType.List(this.member.pythonType())
+        is RustType.HashMap -> PythonType.Dict(this.key.pythonType(), this.member.pythonType())
+        is RustType.HashSet -> PythonType.Set(this.member.pythonType())
+        is RustType.Reference -> this.member.pythonType()
+        is RustType.Option -> PythonType.Optional(this.member.pythonType())
+        is RustType.Box -> this.member.pythonType()
+        is RustType.Dyn -> this.member.pythonType()
+        is RustType.Opaque -> PythonType.Opaque(this.name, this.namespace)
+        // TODO(Constraints): How to handle this?
+        // Revisit as part of https://github.com/awslabs/smithy-rs/issues/2114
```

Comment:
I don't think you have to.. I was talking with @david-perez and right now Python only handles constraints during deserialization and hands you off the original structure, without newtypes if the validation succeeded or an error otherwise.